pkg/providers/snow/snow.go

```diff
@@ -38,19 +37,21 @@ type snowProvider struct {
 	writer                filewriter.FileWriter
 	retrier               *retrier.Retrier
 	bootstrapCreds        bootstrapCreds
+	configManagerEntry    *ConfigManagerEntry
 }
 
 type ProviderKubectlClient interface {
 	DeleteEksaDatacenterConfig(ctx context.Context, snowDatacenterResourceType string, snowDatacenterConfigName string, kubeconfigFile string, namespace string) error
 	DeleteEksaMachineConfig(ctx context.Context, snowMachineResourceType string, snowMachineConfigName string, kubeconfigFile string, namespace string) error
 }
 
-func NewProvider(providerKubectlClient ProviderKubectlClient, writer filewriter.FileWriter, now types.NowFunc) *snowProvider {
+func NewProvider(providerKubectlClient ProviderKubectlClient, writer filewriter.FileWriter, aws aws.Clients, now types.NowFunc) *snowProvider {
```

Comment:
I would pass the maybe the config manager entry as input to this constructor to keep it aware only of its direct dependencies

Reply:
'Inject all the things' with a root initializing constructor